chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -145,14 +145,20 @@ impl EthereumAdapter {
             .await;
 
         // Determine if the provider supports block receipts based on the fetched result.
-        let supports_block_receipts = block_receipts_result
+        let (supports_block_receipts, error_message) = block_receipts_result
             .map(|receipts_option| {
                 // Ensure the result contains non-empty receipts
-                receipts_option.map_or(false, |receipts| !receipts.is_empty())
+                (
+                    receipts_option.map_or(false, |receipts| !receipts.is_empty()),
+                    None,
+                )
             })
-            .unwrap_or(false); // Default to false if there's an error in fetching receipts.
+            .unwrap_or_else(|err| {
+                // Store the error message and default to false
+                (false, Some(err.to_string()))
+            });
 
-        info!(logger, "Checked if provider supports eth_getBlockReceipts"; "provider" => provider, "supports_block_receipts" => supports_block_receipts);
+        info!(logger, "Checked if provider supports eth_getBlockReceipts"; "provider" => provider, "supports_block_receipts" => supports_block_receipts, "error" => error_message.unwrap_or_else(|| "".to_string()));
```

Comment:
Minor nit: it would be nicer in the logs if it said `error => none` if there was no error than just an empty string